src/Workspaces/Core/Portable/FindSymbols/FindReferences/DependentProjectsFinder.cs

```diff
@@ -104,7 +104,15 @@ internal static partial class DependentProjectsFinder
                     continue;
 
                 if (!result.TryGetValue(assembly, out var projectAndVisibility))
-                    projectAndVisibility = (solution.GetProject(assembly, cancellationToken), symbol.GetResultantVisibility());
+                {
+                    // First, if this was a source-symbol try to get the original project for this assembly
+                    // (GetOriginatingProject).  If this is a metadata symbol, also see if we can find the original
+                    // source-project for it (GetProject) which happens in cross language P2P references.
```

Comment:
It might be helpful to expand on this comment and explain why we're doing it here; it's not exactly obvious to me....

Reply:
gotcha.  sorry, i hoped the comment helped, but i can expand it further.  Effectively, GetOriginatingProject is the non-thinking helper that says "you have a source-symbol, here is the literal project it came from".  GetProject is hte one that maps through things like P2P symbols in VB/C#.  So if VB references C# and has a VB metadata symbol for a symbol that was originally in C# source, it finds the original C# source project.

It probably makes sense to make a helper that does all of this at once.